File: Apps/ASPNETCore_MVC/ASPNETCore_MVC/Models/UploadFileFormModel.cs
using System.ComponentModel.DataAnnotations;
using LibServices.Configuration;

namespace ASPNETCore_MVC.Models
{
    /// <summary>
    /// Model representing the form data for file uploads.
    /// </summary>
    public class UploadFileFormModel
    {
        /// <summary>
        /// The file uploaded by the user. This field is required.
        /// </summary>
        [Required(ErrorMessage = "Please select a file.")]
        public IFormFile? FormFile { get; set; }

        /// <summary>
        /// The name of the uploaded file.
        /// </summary>
        public string FileName { get; set; } = string.Empty;

        /// <summary>
        /// The file path where the uploaded file will be saved.
        /// </summary>
        public string FilePath { get; set; } = string.Empty;

        /// <summary>
        /// The folder path where the uploaded file will be saved.
        /// Default value from <see cref="UploadFileSettings"/> if not specified.
        /// </summary>
        public string FilePathTarget { get; set; } = UploadFileSettings.FilePathTarget;

        /// <summary>
        /// The custom file name provided by the user (optional).
        /// Default value from <see cref="UploadFileSettings"/> if not provided.
        /// </summary>
        [Display(Name = "Custom File Name")]
        public string CustomFileName { get; set; } = UploadFileSettings.CustomFileName;
    }
}

File: Apps/AngularWebApp/AngularWebApp/src/app/models/upload-file-form.model.ts
export class UploadFileFormModel {
    /**
     Represents the form file to be uploaded.
     @param FormFile The actual file object selected by the user.
     @param FileName The original name of the file as selected.
     @param CustomFileName The custom name for the file provided by the user, if any.
     @param FilePathTarget The target folder path where the file should be uploaded.
    */
    constructor(
        public FormFile: File,
        public FileName: string,
        public CustomFileName: string,
        public FilePathTarget: string
    ) {}
}
